cli/js/tests/url_test.ts

```diff
@@ -213,6 +213,19 @@ unitTest(function createBadUrl(): void {
   });
 });
 
+unitTest(
+  {
+    // TODO this is standard behavior of URL
+    ignore: true,
+  },
+  function pathnameShouldEncodeComponents() {
+    const url = new URL("http://🦕.com/🦖?q=🎉");
+    assertEquals(url.host, "xn--lt9h.com");
+    assertEquals(url.pathname, "/%F0%9F%A6%96");
+    assertEquals(url.search, "?q=%F0%9F%8E%89");
+  }
+);
+
```

Comment:
This seems unrelated to this PR. Tho I agree we need to get this working. Does this require ICU?

Reply:
Sorry, what do you mean "ICU"?

Reply:
I tired to add tests for `writeRequest` and expected url to be encoded. `url.pathname` must return URL encoded component of paths. May I separate it into another branch?